### ISSUE TITLE
Implement the cl-pretrainer training

### ISSUE THREAD

**Issue by afmjoaa** [nerdslab-club/cl_model#13]
- [x] Implement construct_batches_for_cl_pre_trainer inside the batch builder 
- [x] Implement the training loop and return the batch accuracy and batch loss on each epoch.
- [x] Add checkpoint manager for cl pre trainer model
- [x] Add test for training and saving the model
